src/core/componentSystems/BoundaryVisibilitySystem.js

```diff
@@ -0,0 +1,81 @@
+import { MRSystem } from 'mrjs/core/MRSystem';
+import { MRDivEntity } from 'mrjs/core/MRDivEntity';
+import { MRPanel } from 'mrjs/core/entities/MRPanel';
+
+/**
+ * @function
+ * @description Observe a target MRDivEntity and make the associated object visible only if it is in visible position in a root MRDivEntity
+ * @param {MRDivEntity} root
+ * @param {MRDivEntity} target
+ */
+const observe = (root, target) => {
+    // TODO: Callback is fired asynchronously so no guaranteed to be called immediately when the
+    //       visibility from the layout position changes. Therefore, the visibility of the associated
+    //       Object3D's might be updated a few frames later after when it really need to be. It might
+    //       affect the user experience. Fix it if possible.
+    const observer = new IntersectionObserver(
+        (entries) => {
+            for (const entry of entries) {
+                // TODO: Endure the visibility set by other systems is not overridden. For example,
+                //       even if another system has made an entity visible within the panel for some reasons,
+                //       this system makes it visible. This issue should be fixed.
+                entry.target.object3D.visible = entry.intersectionRatio > 0;
```

Comment:
Any good idea to fix this issue? @michaelthatsit @hanbollar

Reply:
oof. so this overrides the other systems? Calling it earlier in the Systems order might fix it. They're stepped sequentially, so any subsequent system would override this one.

Reply:
As written in the updated inline comment, we may say that the callback is executed before all the sync systems because the callback is fired asynchronously.

But maybe it would be still a bit problematic in some scenarios. For example if a system updates a visibility of an object and want it to have an effect for some frames until certain condition is met, the callback of BoundaryVisibilitySystem can override the visibility.